code/parse/sexp.cpp

```diff
@@ -16016,24 +15997,11 @@ void sexp_destroy_instantly(int n)
 
 void multi_sexp_destroy_instantly()
 {
-	int ship_num;
-	object *ship_obj_p;
+	ship *shipp;
 
 	// destroy ships
-	while (Current_sexp_network_packet.get_ship(ship_num))
-	{
-		// if it still exists, destroy it
-		if (ship_num >= 0)
-		{
-			ship_obj_p = &Objects[Ships[ship_num].objnum];
-
-			// if it's the player don't destroy
-			if (ship_obj_p != Player_obj)
-			{
-				ship_destroy_instantly(ship_obj_p);
-			}
-		}
-	}
+	while (Current_sexp_network_packet.get_ship(shipp))
+		ship_destroy_instantly(&Objects[shipp->objnum]);
```

Comment:
This drops the check for the player ship. Not exactly sure what object that would be in multiplayer but may be relevant here.

Reply:
That check would actually be redundant here.  In multiplayer mode, it's not *the* player, it's *any* player.  But I've added some comments for clarification.